include/boost/json/impl/monotonic_resource.ipp

```diff
@@ -79,35 +97,67 @@ monotonic_resource::
 }
 
 monotonic_resource::
-monotonic_resource() noexcept
-    : block_size_(64 * 1024)
+monotonic_resource(
+    std::size_t block_size,
```

Comment:
I think "block_size" is the wrong parameter. We should track the standard here. Instead of block_size we should make it `initial_size` and let that be the size of the first block allocated. And we should switch around the parameters, `(max_alloc, initial_size)` and provide defaults for both of them, with `initial_size` defaulting to zero (meaning, to use some internal constant). Again we should track the standard here, and the standard doesn't even have `initial_size` so there's that.